triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/process/ledger/LedgerApiClient.scala

```diff
@@ -25,41 +30,109 @@ object LedgerApiClient {
       request: SubmitRequest,
       replyTo: ActorRef[TriggerProcess.Message],
+      timestamp: Long,
+      reportId: UUID,
+      triggerId: UUID,
+      triggerDefRef: Ref.DefinitionRef,
   ) extends Message
 
   def create(
-      client: LedgerClient
+      client: LedgerClient,
+      report: ActorRef[ReportingProcess.Message],
   )(implicit config: TriggerSimulationConfig): Behavior[Message] = {
-    Behaviors.receive { case (context, CommandSubmission(request, replyTo)) =>
-      try {
-        discard(
-          Await.result(
-            client.commandClient.submitSingleCommand(request),
-            config.ledgerSubmissionTimeout,
+    Behaviors.receive {
+      case (
+            context,
+            CommandSubmission(
+              request,
+              replyTo,
+              submissionTimestamp,
+              reportId,
+              triggerId,
+              triggerDefRef,
+            ),
+          ) =>
+        val commandId = request.getCommands.commandId
+        report ! ReportingProcess.SubmissionUpdate(
+          SubmissionReporting.SubmissionUpdate(
+            submissionTimestamp,
+            commandId,
+            reportId,
+            triggerId,
+            triggerDefRef,
+            request,
           )
         )
-      } catch {
-        case cause: StatusRuntimeException if cause.getStatus.getCode != Code.UNAUTHENTICATED =>
-          context.log.info(
-            s"Ledger API encountered a command submission failure for $request - sending completion failure to $replyTo",
-            cause,
+        try {
+          discard(
+            Await.result(
+              client.commandClient.submitSingleCommand(request),
+              config.ledgerSubmissionTimeout,
+            )
           )
-          replyTo ! TriggerProcess.MessageWrapper(
-            TriggerMsg.Completion(
-              Completion(
-                request.getCommands.commandId,
-                Some(Status(cause.getStatus.getCode.value(), cause.getStatus.getDescription)),
+        } catch {
+          case cause: StatusRuntimeException if cause.getStatus.getCode != Code.UNAUTHENTICATED =>
+            context.log.info(
+              s"Ledger API encountered a command submission failure for $request - sending completion failure to $replyTo",
+              cause,
+            )
+            val completionTimestamp = System.currentTimeMillis()
+            val completion = Completion(
+              request.getCommands.commandId,
+              Some(Status(cause.getStatus.getCode.value(), cause.getStatus.getDescription)),
+            )
+            replyTo ! TriggerProcess.MessageWrapper(TriggerMsg.Completion(completion))
+            report ! ReportingProcess.SubmissionUpdate(
+              SubmissionReporting.CompletionUpdate(
+                completionTimestamp,
+                commandId,
+                triggerId,
+                triggerDefRef,
+                completion,
```

Comment:
When command submissions failure (as part of the ledger request), we need to match real trigger behaviour and generate a "fake" completion failure signal.